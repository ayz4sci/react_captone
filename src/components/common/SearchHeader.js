import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import InputAdornment from '@material-ui/core/InputAdornment';
import Search from '@material-ui/icons/Search';
import LoadingDots from './LoadingDots'

class SearchHeader extends React.Component{
    render(){
        return (
            // 
            <div className="row justify-content-md-center">
                <span className="col-4 m-4 header">
                    <TextField
                        id="search"
                        label="Git Cheat Sheet"
                        variant="outlined"
                        onChange={this.props.onSearchChange}
                        InputProps={{
                            endAdornment: (
                            <InputAdornment position="end">
                                <Search />
                            </InputAdornment>
                            ),
                        }} />
                    {  this.props.loading && <LoadingDots className="m-1" interval={400} dots={4} /> }
                </span>
            </div>
        )
    }
}

SearchHeader.propType = {
    loading: PropTypes.bool.isRequired,
    onSearchChange: PropTypes.func.isRequired
}

export default SearchHeader;