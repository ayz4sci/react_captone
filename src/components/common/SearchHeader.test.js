import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchHeader from './SearchHeader';

it('renders correctly', () => {
    const props = { 
        loading: false,
        onSearchChange: jest.fn()
    };

    const wrapper = shallow(<SearchHeader {...props} />, {mode: 'shallow'})
    expect(toJson(wrapper)).toMatchSnapshot();
});