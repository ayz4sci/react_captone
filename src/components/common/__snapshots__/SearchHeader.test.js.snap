// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row justify-content-md-center"
>
  <span
    className="col-4 m-4 header"
  >
    <TextField
      InputProps={
        Object {
          "endAdornment": <WithStyles(InputAdornment)
            position="end"
          >
            <pure(SearchIcon) />
          </WithStyles(InputAdornment)>,
        }
      }
      id="search"
      label="Git Cheat Sheet"
      onChange={[MockFunction]}
      required={false}
      select={false}
      variant="outlined"
    />
  </span>
</div>
`;
